### PR TITLE
WINC-614: Remove test overlap in upgrade suite

### PR DESCRIPTION
**Pull request by saifshaikh48**
This PR aims to reduce the amount of time our end-to-end tests take to complete by dropping a redundant case from the upgrade tests and spreading platform-agnostic tests across cloud providers in CI.

### CODE REVIEW

File: test/e2e/upgrade_test.go
Comment (saifshaikh48) [inline test/e2e/upgrade_test.go:76]: Reviewers, as part of the individual test suites (network, upgrade, metrics, etc) within the e2e tests, which route should we take?
1. just assume that the cluster is in testable state (e.g. operator is up and, for this suite, correct # nodes initially have valid version annotation) -- this approach calls for the removal of this setup method
2. have guards to check, else fail fast/don't even run test case
3. try to coerce cluster to reach proper state

File: test/e2e/upgrade_test.go
Comment (saifshaikh48) [inline test/e2e/upgrade_test.go:77]: Example of option 2, guard + fail fast

File: test/e2e/upgrade_test.go
Comment (saifshaikh48) [inline test/e2e/upgrade_test.go:86]: Example of option 3, manual coercion

File: test/e2e/upgrade_test.go
Comment (saifshaikh48) [inline test/e2e/upgrade_test.go:76]: Reviewers, as part of the individual test suites (network, upgrade, metrics, etc) within the e2e tests, which route should we take?
1. just assume that the cluster is in testable state (e.g. operator is up and, for this suite, correct # nodes initially have valid version annotation) -- this approach calls for the removal of this setup method
2. have guards to check, else fail fast/don't even run test case
3. try to coerce cluster to reach proper state

File: test/e2e/upgrade_test.go
Comment (saifshaikh48) [inline test/e2e/upgrade_test.go:84]: Example of option 2, guard + fail fast

File: test/e2e/upgrade_test.go
Comment (saifshaikh48) [inline test/e2e/upgrade_test.go:88]: Example of option 3, manual coercion

**Comment by mansikulkarni96** [comment]
changes LGTM will /lgtm once approved.

File: test/e2e/upgrade_test.go
Comment (sebsoto) [inline test/e2e/upgrade_test.go:55]: In this commit, this line + the deletion of the function are really the only changes that are accomplishing the goals you outlined in the commit message.
Everything you've done is beneficial for the codebase as a whole, but it would be better to separate these related small quality of life refactors into a separate commit, for the sake of reviewers. It makes it easier for us to see what should have a functional impact, and what shouldn't :)

**Review by sebsoto — CHANGES_REQUESTED** [review]
The work done here LGTM but I'd prefer to see the commit separation done, not a blocker from me though.

File: test/e2e/upgrade_test.go
Comment (saifshaikh48) [reply test/e2e/upgrade_test.go:55]: ACK, broke it out into separate commits.

**Comment by saifshaikh48** [comment]
> changes LGTM will /lgtm once approved.

@mansikulkarni96 this has been updated & approved, PTAL

**Pull request merged**
